src/builder_state.rs

```diff
@@ -653,6 +636,33 @@ impl<TYPES: NodeType> BuilderProgress<TYPES> for BuilderState<TYPES> {
     #[tracing::instrument(skip_all, name = "build block",
                                     fields(builder_built_from_proposed_block = %self.built_from_proposed_block))]
     async fn build_block(&mut self, matching_vid: VidCommitment) -> Option<BuildBlockInfo<TYPES>> {
+        // provide atleast 1 txn inside the block process the transactions
+        // ideally this should be a threshold should a configurable value, if it gets non-zero before time return, otherwise return after timeout
+        let timeout_after = Instant::now() + self.maximize_txn_capture_timeout;
+        while let Ok(tx) = self.tx_receiver.try_recv() {
```

Comment:
We should only do this if we don't already have any (or, later, the threshold) transactions when `build_block` gets called. This is, otherwise, great, but it shouldn't enter this if `self.tx_hash_to_available_txns.len() > 0`.

Reply:
https://github.com/EspressoSystems/hotshot-builder-core/pull/102/commits/deb63a31b5c18d82963acc8821de611c1241aa63